source/server/lsp-servers.ads

```diff
@@ -50,24 +54,80 @@ package LSP.Servers is
       Params   : LSP.Messages.ApplyWorkspaceEditParams;
       Applied  : out Boolean;
       Error    : out LSP.Messages.Optional_ResponseError);
+   --  ??? Needs doc
+   --  ??? In the current implementation, Applied is always True and
+   --  Error is always unset.
 
 private
 
+   -------------------------
+   --  Tasking in the ALS --
+   -------------------------
+
+   --  The server has 3 tasks:
+   --    The input task (currently: the main thread)
+   --         This reads input coming from stdin, forms requests, and places
+   --         them on the requests queue.
+   --    The processing task:
+   --         This is the task where libadalang lives. This task receives
+   --         requests from the request queue, processes them, and returns
+   --         the responses from them on the output queue.
+   --    The output task:
+   --         This task reads the responses coming from the output queue,
+   --         and writes them to the standard output.
+
+   type Stream_Access is access all Ada.Streams.Root_Stream_Type'Class;
+
+   package Unbounded_String_Queue_Interface is new
+     Ada.Containers.Synchronized_Queue_Interfaces
+       (Ada.Strings.Unbounded.Unbounded_String);
+   package Requests_Queues is new
+     Ada.Containers.Unbounded_Synchronized_Queues
+       (Unbounded_String_Queue_Interface);
+   package Output_Queues is new
+     Ada.Containers.Unbounded_Synchronized_Queues
+       (Unbounded_String_Queue_Interface);
+
+   type Requests_Queue_Access is access Requests_Queues.Queue;
+   type Output_Queue_Access is access Output_Queues.Queue;
+
+   --  The processing task
+   task type Processing_Task_Type is
+      entry Start
+        (In_Queue     : Requests_Queue_Access;
```

Comment:
Have you considered an idea to make queue as discriminant of the type instead of passing it in the Initialize call?